src/main/scala/org/camunda/feel/syntaxtree/Val.scala

```diff
@@ -287,6 +288,10 @@ case class ValError(error: String) extends Val {
   override def toString: String = s"error(\"$error\")"
 }
 
+case class ErroredVal(value: Val, errorType: EvaluationFailureType)  extends Val {
+  override def toString: String = value.toString
+}
```

Comment:
:x: The new type is only used in combination with `ValNull`. It seems like the failure type should be an additional property of `ValNull`. And, it is used only within the interpreter.